modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/elasticsearch/snapshot/SnapshotServiceImpl.java

```diff
@@ -96,15 +98,29 @@ public RestoreResult restore( final RestoreParams restoreParams )
     private RestoreResult doRestore( final RestoreParams restoreParams )
     {
         checkSnapshotRepository();
-        validateSnapshot( restoreParams.getSnapshotName() );
+
+        String snapshotName;
+
+        if ( restoreParams.isLatest() )
+        {
+            final SnapshotResult snapshotResult = Iterables.getLast( list() );
```

Comment:
We are slowly getting away from Guava. We don't use Iterables class (almost) anywhere else. 
`list.get( list.size() - 1 )` is not perfect but good enough alternative.

It would be also good to check if size is not 0 and throw exception with clear message: "No snapshots found" for ex.

Reply:
solved